src/compiler/checker.ts

```diff
@@ -30477,6 +30487,22 @@ namespace ts {
             return Debug.assertNever(node.keywordToken);
         }
 
+        function checkMetaPropertyKeyword(node: MetaProperty): Type {
+            switch (node.keywordToken) {
+                case SyntaxKind.ImportKeyword:
+                    return getGlobalImportMetaExpressionType();
+                case SyntaxKind.NewKeyword:
+                    const targetSymbol = getSymbolAtLocation(node);
+                    if (targetSymbol) {
+                        const type = getTypeOfSymbolAtLocation(targetSymbol, node);

```

Comment:
Presumably this just goes through the pipes until it hits `checkNewTargetMetaProperty`, right? I think it’s probably alright to just call that directly, though I don’t think it matters too much.

Reply:
It is possible that it will be `errorType` if it’s being written outside a valid container, in which case maybe you just want to return `errorType` as well?

Reply:
Good idea and catch! Fixed in e8030a55b30da1a6e690425637316ea3813a265c